include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,226 @@
+      // Copy V panel into VV
+      copySingleTile(mat_v.read(LocalTileIndex(i, k)), mat_vv(LocalTileIndex(i, 0)));
+
+      // Setting VV
+      auto tile_i0 = mat_vv(LocalTileIndex(i, 0));
+      if (i <= k) {
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::set0<T>), std::move(tile_i0));
+      }
+      else if (i == k + 1) {
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      std::move(tile_i0));
+      }
+
+      // Copy VV into W
+      copySingleTile(mat_vv.read(LocalTileIndex(i, 0)), mat_w(LocalTileIndex(i, 0)));
+    }
+
+    // Reset W2 to zero
+    setZero(mat_w2);
+
+    // TODO: instead of using a full matrix, choose a "column" matrix. The problem is that last tile
+    // should be square but may have different size.
+    const GlobalTileIndex v_start{k + 1, k};
+    auto taus_panel = taus[k];
+    const SizeType taupan = (is_last) ? last_mb : mat_v.blockSize().cols();
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(taupan, mat_v, v_start, taus_panel,
+                                                               mat_t(LocalTileIndex{k, k}));
+
+    for (SizeType i = k + 1; i < m; ++i) {
+      auto kk = LocalTileIndex{k, k};
+      // WH = V T
+      auto ik = LocalTileIndex{i, 0};
+      hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_t = mat_t.read(kk);
+      auto tile_w = mat_w(ik);
+
+      if (mat_t.tileSize(GlobalTileIndex{k, k}).rows() != mat_w.tileSize(GlobalTileIndex{i, 0}).cols()) {
+        TileElementIndex origin(0, 0);
+        TileElementSize size(mat_t.tileSize(GlobalTileIndex{k, k}).rows(),
+                             mat_t.tileSize(GlobalTileIndex{k, k}).cols());
+        const matrix::SubTileSpec spec({origin, size});
+        auto subtile_w = splitTile(tile_w, spec);
+        trmmPanel(executor_np, tile_t, std::move(subtile_w));
+      }
+      else
+        trmmPanel(executor_np, mat_t.read(kk), std::move(tile_w));
+    }
+
+    for (SizeType j = 0; j < n; ++j) {
+      auto kj = LocalTileIndex{0, j};
+      for (SizeType i = k + 1; i < m; ++i) {
+        auto ik = LocalTileIndex{i, 0};
+        auto ij = LocalTileIndex{i, j};
+        // W2 = W C
+        gemmUpdateW2(executor_np, mat_w(ik), mat_c.read(ij), mat_w2(kj));
+      }
+    }
+
+    for (SizeType i = k + 1; i < m; ++i) {
+      auto ik = LocalTileIndex{i, 0};
+      for (SizeType j = 0; j < n; ++j) {
+        auto kj = LocalTileIndex{0, j};
+        auto ij = LocalTileIndex{i, j};
+        // C = C - V W2
+        gemmTrailingMatrix(executor_np, mat_vv.read(ik), mat_w2.read(kj), mat_c(ij));
+      }
+    }
+  }
+}
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  DLAF_UNIMPLEMENTED(grid);
+}

```

Comment:
You decided to anticipate in this PR about local implementaiton also the distributed signature, but this will end up in unused parameters.